src/linker/Linker.Steps/IPerAssemblyStep.cs

```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Mono.Cecil;
+
+namespace Mono.Linker.Steps
+{
+
+	public interface IPerAssemblyStep
+	{
+		void Initialize (LinkContext context);
+		void ProcessAssembly (AssemblyDefinition assembly);
```

Comment:
I think it's worth it to add comments on the intended usage and when is this called. We will eventually have to document it for real, but for now comments in code should suffice.